static/js/reducers/ui.js

```diff
@@ -120,131 +120,153 @@ export const INITIAL_UI_STATE: UIState = {
-    return Object.assign({}, state, {
-      educationLevelAnswers: action.payload
-    });
+    return {
+      ...state,
+      educationLevelAnswers: action.payload,
+    };
   case CLEAR_UI:
     return INITIAL_UI_STATE;
   case SET_USER_PAGE_DIALOG_VISIBILITY: {
-    return Object.assign({}, state, {
-      userPageDialogVisibility: action.payload
-    });
+    return {
+      ...state,
+      userPageDialogVisibility: action.payload,
+    };
   }
   case SET_USER_PAGE_ABOUT_ME_DIALOG_VISIBILITY: {
-    return Object.assign({}, state, {
-      userPageAboutMeDialogVisibility: action.payload
-    });
+    return {
+      ...state,
+      userPageAboutMeDialogVisibility: action.payload,
+    };
   }
   case SET_SHOW_EDUCATION_DELETE_DIALOG: {
-    return Object.assign({}, state, {
-      showEducationDeleteDialog: action.payload
-    });
+    return {
+      ...state,
+      showEducationDeleteDialog: action.payload,
+    };
   }
   case SET_SHOW_WORK_DELETE_DIALOG: {
-    return Object.assign({}, state, {
-      showWorkDeleteDialog: action.payload
-    });
+    return {
+      ...state,
+      showWorkDeleteDialog: action.payload,
+    };
   }
   case SET_DELETION_INDEX: {
-    return Object.assign({}, state, {
-      deletionIndex: action.payload
-    });
+    return {
+      ...state,
+      deletionIndex: action.payload,
+    };
   }
   case SET_PROFILE_STEP: {
-    return Object.assign({}, state, {
-      profileStep: action.payload
-    });
+    return {
+      ...state,
+      profileStep: action.payload,
+    };
   }
   case SET_USER_MENU_OPEN: {
-    return Object.assign({}, state, {
-      userMenuOpen: action.payload
-    });
+    return {
+      ...state,
+      userMenuOpen: action.payload,
+    };
   }
   case SET_SEARCH_FILTER_VISIBILITY: {
-    return Object.assign({}, state, {
-      searchFilterVisibility: action.payload
-    });
+    return {
+      ...state,
+      searchFilterVisibility: action.payload,
+    };
   }
   case SET_EMAIL_DIALOG_VISIBILITY: {
-    return Object.assign({}, state, {
-      emailDialogVisibility: action.payload
-    });
+    return {
+      ...state,
+      emailDialogVisibility: action.payload,
+    };
   }
   case SET_ENROLL_DIALOG_ERROR: {
-    return Object.assign({}, state, {
-      enrollDialogError: action.payload
-    });
+    return {
+      ...state,
+      enrollDialogError: action.payload,
+    };
   }
   case SET_TOAST_MESSAGE: {
-    return Object.assign({}, state, {
-      toastMessage: action.payload
-    });
+    return {
+      ...state,
+      toastMessage: action.payload,
+    };
   }
   case SET_ENROLL_SELECTED_PROGRAM: {
-    return Object.assign({}, state, {
-      enrollSelectedProgram: action.payload
-    });
+    return {
+      ...state,
+      enrollSelectedProgram: action.payload,
+    };
   }
   case SET_ENROLL_DIALOG_VISIBILITY: {
-    return Object.assign({}, state, {
+    return {
+      ...state,
       enrollDialogVisibility: action.payload
-    });
+    };
```

Comment:
I have a maybe silly idea - what if for this file we defined a function like this:

```js
const mergeState = R.curry((state, diff) => (
  { ...state, ...diff }
));
```

and then above our `switch` block we'd do something like 

```js
const mergeCurState = mergeState(state);
```

then for each case our return values would look like:

```js
return mergeCurState({ enrollDialogVisibility: action.payload });
```

just a thought! might not be more readable that way.